core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
My thoughts was that by only allowing a Boolean flag a spark job either uses all GPUs from a host or not, which it won't be able to have different GPu devices shared by different jobs. By specifying a limit at least there is ability to let a job specify how much GPUs it should grab per node. thoughts?